src/resolve/resolved-dns-answer.c

```diff
@@ -685,6 +686,19 @@ int dns_answer_reserve(DnsAnswer **a, size_t n_free) {
                         return -ENOMEM;
 
                 n->n_allocated = ns;
+
+                set_clear(n->set_items);
```

Comment:
any particular reason to clear and repopulate, rather than just increasing the size as needed?

Reply:
@bluca I think the set contains pointers into array `items` at the end of DnsAnswer structure. Once the structure is realloc()ated, the pointers still refer to old area, so the set has to be refreshed with new pointers to items. So increasing the size would not work.

Reply:
correct

Reply:
Thanks! Missed that detail